src/uct/ib/rc/base/rc_ep.h

```diff
@@ -419,18 +419,14 @@ uct_rc_fc_req_moderation(uct_rc_fc_t *fc, uct_rc_iface_t *iface)
 static UCS_F_ALWAYS_INLINE int
 uct_rc_ep_atomic_fence(uct_rc_iface_t *iface, uct_ib_fence_info_t* fi, int flag)
 {
-    int fence = fi->fence_flag;
+    int fence;
 
     /* a call to iface_fence increases beat, so if endpoint beat is not in
      * sync with iface beat it means the endpoint did not post any WQE with
      * fence flag yet */
-    if (fi->fence_beat != iface->tx.fi.fence_beat) {
-        fi->fence_beat = iface->tx.fi.fence_beat;
-        fence |= iface->tx.fi.fence_flag;
-    }
-
-    fi->fence_flag = 0;
-    return fence ? flag : 0;
+    fence          = flag * !!(fi->fence_beat != iface->tx.fi.fence_beat);
```

Comment:
why `!!` is needed? `(fi->fence_beat != iface->tx.fi.fence_beat)` already returns result of boolean operation
could you document meaning of `flag`? it's not obvious from arithmetics

Reply:
!! used to be sure that result is 1 or 0
flag is incoming parameter returned in case if fence requested

Reply:
ok, fixed